src/lib/transaction/transactionGroupHelper.ts

```diff
@@ -0,0 +1,51 @@
+import type { IGroupsFromParam, ITransactionGroup } from './ITxParams';
+import type { OperationComponentTypes } from './operations/OperationComponentTypes';
+
+export default function groupComponents(
+    operations: typeof OperationComponentTypes[],
+    groups: IGroupsFromParam[],
+): (typeof OperationComponentTypes | ITransactionGroup)[] {
+    const lastGroup = groups[groups.length - 1];
+    const transactionGroups: (typeof OperationComponentTypes | ITransactionGroup)[] = [];
+
+    if (groups.length === 0) {
+        console.log("A group of operations wasn't provided");
+        return operations;
+    } else if (lastGroup && !operations[lastGroup.to]) {
+        console.error('There are fewer operations than the groups says');
+        return operations;
+    } else {
+        let startIndex = 0;
+
+        for (let i = 0; i < groups.length; i++) {
+            const currentGroup = groups[i];
+            const nextGroup = groups[i + 1];
+
+            if (nextGroup && currentGroup && currentGroup.from > nextGroup.from) {
+                console.error('The group object is not well sorted');
```

Comment:
Please make this message give a more thorough explanation of what went wrong, for instance include which groups are being evaluated and how did it come to the conclusion that they are not well sorted